src/main/java/gov/nasa/pds/validate/report/FullReport.java

```diff
@@ -54,91 +52,157 @@
-    // validation,
-    // we are now into integrity checks.
-    if (title.toLowerCase().contains("pds4 bundle")
-        || title.toLowerCase().contains("pds4 collection")) {
-      this.integrityCheckFlag = true;
-    }
-
-    writer.println();
-    writer.println();
-    writer.println(title);
+  protected void append(Status status, String target) {
+    this.currentStatus = status;
+    this.currentTarget = target;
+    this.getWriter().println();
+    this.getWriter().print("  ");
+    this.getWriter().print(status.getName());
+    this.getWriter().print(": ");
+    this.getWriter().println(target);
   }
 
   @Override
-  protected void printRecordMessages(PrintWriter writer, Status status, URI sourceUri,
-      List<ValidationProblem> problems) {
-    Map<String, List<ValidationProblem>> externalProblems = new LinkedHashMap<>();
-    Map<String, List<ContentProblem>> contentProblems = new LinkedHashMap<>();
-    writer.println();
-    writer.print("  ");
-    writer.print(status.getName());
-    writer.print(": ");
-    writer.println(sourceUri.toString());
+  protected void append(String title) {
+    this.getWriter().println();
+    this.getWriter().println(title);
+  }
 
-    // Print all the sources problems and gather all external problems
-    for (Iterator<ValidationProblem> iterator = problems.iterator(); iterator.hasNext();) {
-      ValidationProblem problem = iterator.next();
-      if (problem instanceof ContentProblem) {
-        ContentProblem contentProb = (ContentProblem) problem;
-        List<ContentProblem> contentProbs = contentProblems.get(contentProb.getSource());
-        if (contentProbs == null) {
-          contentProbs = new ArrayList<>();
-        }
-        contentProbs.add(contentProb);
-        contentProblems.put(contentProb.getSource(), contentProbs);
-      } else if (((problem.getTarget() == null)) || (problem.getTarget().getLocation() == null)
-          || sourceUri.toString().equals(problem.getTarget().getLocation())) {
-        printProblem(writer, problem);
-      } else {
-        List<ValidationProblem> extProbs = externalProblems.get(problem.getTarget().getLocation());
-        if (extProbs == null) {
-          extProbs = new ArrayList<>();
+  @Override
+  protected void append(ValidationProblem problem) {
+    if (this.currentStatus == Status.SKIP) {
+      this.getWriter().print("      ");
+      String severity = "";
+      if (problem.getProblem().getSeverity() == ExceptionType.FATAL) {
+        severity = "FATAL_ERROR";
+      } else if (problem.getProblem().getSeverity() == ExceptionType.ERROR) {
+        severity = "ERROR";
+      } else if (problem.getProblem().getSeverity() == ExceptionType.WARNING) {
+        severity = "WARNING";
+      } else if (problem.getProblem().getSeverity() == ExceptionType.INFO) {
+        severity = "INFO";
+      } else if (problem.getProblem().getSeverity() == ExceptionType.DEBUG) {
+        severity = "DEBUG";
+      }
```

Comment:
Should we have a final `else` in case the severity doesn't match any of the conditions? (Or default `severity` to `unknown` instead of empty string?)

Reply:
To really fix it, change ExceptionType.toString() to do the right thing so that we can get rid of the if block. Can do and it may already be that way but this code does not use it and I just did a gut-n-paste from the old stuff since it "works".

Reply:
@nutjob4life 

You sure? I have the spare time. Do want to start a new big job but little clean up is good. Still fixing cucumber too.

Reply:
@nutjob4life 

Yeah, my memory was right and getName() was already implemented correctly. Done.